File: frontend/assets/ts/core/security/protocols.ts
/* SoAI - Shared security protocols [frontend/assets/ts/core/security/protocols.ts] */
// SPDX-License-Identifier: LicenseRef-SoAI-Source-1.0

import type { SanitizeUrlOptions } from '@core/security/urlSanitizer.ts';

interface EscapeSanitizerApi {
    escapeHtml: <T>(value: T) => string;
    escapeAttribute: <T>(value: T) => string;
    sanitizeUrl: <T>(value: T, options?: SanitizeUrlOptions) => string | null;
    sanitizeAbsoluteHttpUrl: <T>(value: T) => string | null;
    resolveHttpsWebsiteUrl: <T>(value: T) => string | null;
}

export type { EscapeSanitizerApi };

File: backend/core/browser_adblock/protocols.py
"""SoAI - Shared browser adblock protocols [backend/core/browser_adblock/protocols.py]"""
# SPDX-License-Identifier: LicenseRef-SoAI-Source-1.0

from __future__ import annotations

from typing import Protocol

from core.browser_adblock.matcher import EasyListMatchResult

__all__ = ("EasyListAdblockServiceProtocol",)


class EasyListAdblockServiceProtocol(Protocol):
    @property
    def next_refresh_unix(self) -> float: ...

    @property
    def source_name(self) -> str: ...

    def match(
        self,
        request_url: str,
        *,
        request_host: str | None,
        request_type: str | None,
        document_host: str | None,
        is_third_party: bool | None,
    ) -> EasyListMatchResult: ...

    async def refresh_from_remote(self) -> None: ...
